presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileWriter.java

```diff
@@ -0,0 +1,407 @@
+
+        private void writeBlock(Block block)
+                throws IOException
+        {
+            checkArgument(!columnClosed, "Column is closed");
+            columnEncoding.encodeColumn(block, output, encodeOutput);
+        }
+
+        public void closeColumn()
+                throws IOException
+        {
+            checkArgument(!columnClosed, "Column is not open");
+
+            encodeOutput.flush();
+            output.close();
+
+            columnClosed = true;
+        }
+
+        public int getBufferedSize()
+        {
+            return lengthOutput.size() + output.size();
+        }
+
+        public Slice getLengthData()
+        {
+            checkArgument(columnClosed, "Column is open");
+            return lengthOutput.slice();
+        }
+
+        public int getUncompressedSize()
+        {
+            checkArgument(columnClosed, "Column is open");
+            return output.size();
+        }
+
+        public int getCompressedSize()
+        {
+            checkArgument(columnClosed, "Column is open");
+            return output.getCompressedSize();
+        }
+
+        public List<Slice> getCompressedData()
+        {
+            checkArgument(columnClosed, "Column is open");
+            return output.getCompressedSlices();
+        }
+
+        public void reset()
+                throws IOException
+        {
+            checkArgument(columnClosed, "Column is open");
+            lengthOutput.reset();
+
+            output = output.createRecycledCompressedSliceOutput();
+            encodeOutput = new ColumnEncodeOutput(lengthOutput, output);
+
+            columnClosed = false;
+        }
+
+        public void destroy()
+        {
+            output.destroy();
+        }
+
+        public long getRetailedSizeInBytes()
+        {
+            return lengthOutput.getRetainedSize() + output.getRetainedSize();
+        }
+
+        private static class ColumnEncodeOutput
+                implements EncodeOutput
+        {
+            private final SliceOutput lengthOutput;
+            private final SliceOutput valueOutput;
+            private int previousOffset;
+            private int previousLength;
+            private int runLength;
+
+            public ColumnEncodeOutput(SliceOutput lengthOutput, SliceOutput valueOutput)
+            {
+                this.lengthOutput = lengthOutput;
+                this.valueOutput = valueOutput;
+                this.previousOffset = valueOutput.size();
+                this.previousLength = -1;
+            }
+
+            @Override
+            public void closeEntry()
+            {
+                int offset = valueOutput.size();
+                int length = offset - previousOffset;
+                previousOffset = offset;
+
+                if (length == previousLength) {
+                    runLength++;
+                }
+                else {
+                    if (runLength > 0) {
+                        int value = ~(runLength);
```

Comment:
I find `~(x)` quite strange. I would write `~x`. But if that's what you prefer, I'm ok.